packages/core/test/foundry/data-verification-mechanism/fork-tests/FakeLifeCycle.ft.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "./CommonDataVerificationMechanismTest.sol";
+
+contract FakeLifeCycle is CommonDataVerificationMechanismTest {
+    function setUp() public {
+        _commonSetup();
+    }
+
+    function test_CanEnqueueRequestAndVoteWithNewlyStakedTokens() public {
+        if (!shouldRunTest) return; // Exit early if we are not executing forked tests.
+
+        // Submit a new request and show it increments.
+        // Ensure we are at the start of a voting round so we can stake and vote without the stake being disabled.
+        if (voting.getVotePhase() == VotingV2Interface.Phase.Reveal) moveToNextRound();
+
+        uint256 numberRequestsPreRequest = voting.getNumberOfPriceRequests();
+        vm.prank(registeredRequester);
+        voting.requestPrice(identifier, requestTime, ancillaryData);
+        assert(voting.getNumberOfPriceRequests() == numberRequestsPreRequest + 1);
+
+        // Mint fresh UMA and stake them.
+        vm.prank(address(voting));
+        votingToken.mint(TestAddress.account1, gatMeetingAmountOfTokens);
+
+        vm.startPrank(TestAddress.account1);
+        votingToken.approve(address(voting), gatMeetingAmountOfTokens);
+        voting.stake(gatMeetingAmountOfTokens);
+        assert(voting.getVoterStakePostUpdate(TestAddress.account1) == gatMeetingAmountOfTokens);
+
+        // Move to next round and vote from the newly staked account.
+        moveToNextRound();
+        int256 price = 1e18;
+        int256 salt = 42069;
+        uint256 roundId = voting.getCurrentRoundId();
+        address account = TestAddress.account1;
+        bytes32 hash =
+            keccak256(abi.encodePacked(price, salt, account, requestTime, ancillaryData, roundId, identifier));
+        voting.commitVote(identifier, requestTime, ancillaryData, hash);
+        moveToNextPhase();
+        voting.revealVote(identifier, requestTime, price, ancillaryData, salt);
+
+        // Check the price has resolved correctly.
+        moveToNextRound();
+        vm.stopPrank();
+        vm.prank(registeredRequester);
+        assertEq(voting.getPrice(identifier, requestTime, ancillaryData), price);
+
+        // Can get rewards (time elapsed) and should be able to unstake with more than started with due to + slashing.
+        vm.startPrank(TestAddress.account1);
+        uint256 stakerBalanceBeforeRewardWithdrawal = votingToken.balanceOf(TestAddress.account1);
+        voting.withdrawRewards();
+        uint256 stakerBalanceAfterRewardWithdrawal = votingToken.balanceOf(TestAddress.account1);
+        assert(stakerBalanceAfterRewardWithdrawal > stakerBalanceBeforeRewardWithdrawal);
```

Comment:
nit: could be a stronger test if we can change this and the assertion on line 59 to equality assertions rather than inequality to be sure the value is exactly what we expect.

For instance, if the contract was broken and sent you 10x the tokens in rewards or unstakes, these checks would all pass.

Reply:
Agree with @mrice32 suggestion

Reply:
done.